src/app/task.service.ts

```diff
@@ -72,8 +72,8 @@ export class TaskService {
       });
   }
 
-  private loadTasksFromAPI() {
-    this.http.get<TaskList>("http://localhost:3001/allTasks/")
+  loadTasksFromAPI() {
+    this.http.get<TaskList>(this.url + "/allTasks/")
       .pipe(
         catchError((error) => {
           console.error('Error fetching tasks from API:', error);
```

Comment:
you still subscribe to the call in the service, please avoid that, instead you can subscribe (and handle unsubscribing) on the component level